src/components/PostDetails.tsx

```
@@ -2,91 +2,58 @@
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
-    setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+  useEffect(() => {
+    if (post) {
+      dispatch(commentsActions.loadCommentsOfPost(post.id));
     }
-  };
+  }, [post.id, dispatch]);
 
-  useEffect(() => {
-    loadComments();
-  }, []);
+  const addComment = async ({ name, email, body }: CommentData) => {
+    const tempId = Date.now();
+    const newComment = {
+      id: tempId,
+      name,
+      email,
+      body,
+      postId: post.id,
+    };
 
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+    dispatch(commentsActions.addCommentLocally(newComment));
 
-  const addComment = async ({ name, email, body }: CommentData) => {
     try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
+      const savedComment = await dispatch(
+        commentsActions.addCommentsOfPost({
+          name,
+          email,
+          body,
+          postId: post.id,
+        }),
+      ).unwrap();
+
+      dispatch(commentsActions.deleteCommentLocally(tempId));
+      dispatch(commentsActions.addCommentLocally(savedComment));
+    } catch (err) {}
```

The `catch` block is empty, which might lead to silent failures. Consider logging the error or providing user feedback to handle errors more effectively.